### ISSUE TITLE
`UnusedVariable` reported when set in different (potentially unreachable) code path

### ISSUE THREAD

**Issue by thunderer** [vimeo/psalm#8036]
When a variable is set in an `if` block and then checked via `isset` outside of that block, latest Psalm reports `UnusedVariable`. For some reason, this violation can't be suppressed, as shown in [this example](https://psalm.dev/r/d6df0c0c82):

```php
function bar(bool $sth): void
{
    if($sth) {
        /** @psalm-suppress UnusedVariable */
        $seen = true;
    }

    if(!isset($seen)) {
        /** @psalm-suppress UnusedVariable */
        $seen = false;
    }
}
```

**Comment by AndrolGenhald** [comment]
Duplicate of #7799, for now the way Psalm is built makes it difficult to get this to work, [rename the variable](https://psalm.dev/r/fffa4bdd84) to start with an underscore.

Also, I do not consider this a false positive, `$seen` _is_ unused, without it being used later the code is functionally equivalent with the second `if` removed. [If it's used](https://psalm.dev/r/b43de102f2) it works fine (although in this case there are false positives because Psalm doesn't work well with variables defined in `if`s, but there's some ongoing effort to improve that; in the meantime [this](https://psalm.dev/r/8032bfeb2a) and similar patterns work fine).
